### PR TITLE
Edit security-proactive-authentication.adoc

### PR DESCRIPTION
**Pull request by rolfedh**
References https://issues.redhat.com/browse/QDOCS-563

### CODE REVIEW

File: docs/src/main/asciidoc/security-proactive-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-proactive-authentication.adoc:45]: `CDI` is a such a well known term that it is probably not an acronym anymore, so I'm not sure  providing  `Contexts and Dependency Injection` for completeness/clarification is needed, I saw it was done in one of your PRs, which was fine, but we may have many more `CDI` acronyms around, I'd consider leaving it as it is

Comment (rolfedh) [reply docs/src/main/asciidoc/security-proactive-authentication.adoc:45]: Okay. Thanks. I'll add this to a list of acronyms that don't need expansion in Quarkus docs.

Comment (rolfedh) [reply docs/src/main/asciidoc/security-proactive-authentication.adoc]: I created an issue for this: https://github.com/quarkusio/quarkus/issues/37867

File: docs/src/main/asciidoc/security-proactive-authentication.adoc
Comment (rolfedh) [inline docs/src/main/asciidoc/security-proactive-authentication.adoc:45]: Does it helpt to rephrase this? 
Suggestion A:
```suggestion
If a non-void secured method returns a value synchronously and proactive authentication is disabled, xref:security-authorize-web-endpoints-reference.adoc#standard-security-annotations[standard security annotations] on CDI beans are not supported on an I/O thread because they must access `SecurityIdentity`.
```
Suggestion B:
```suggestion
When proactive authentication is disabled, xref:security-authorize-web-endpoints-reference.adoc#standard-security-annotations[standard security annotations] used on CDI beans do not function on an I/O thread if a secured method that is not void synchronously returns a value. This limitation arises from the necessity for these methods to access `SecurityIdentity`.
```
What do you think, @sberyozkin ?

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-proactive-authentication.adoc:45]: Hi @rolfedh, suggestion B reads easier to me, thanks

Comment (rolfedh) [reply docs/src/main/asciidoc/security-proactive-authentication.adoc]: Updated. Thanks!

File: docs/src/main/asciidoc/security-proactive-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-proactive-authentication.adoc:129]:
```suggestion
For example, `io.quarkus.oidc.runtime.CodeAuthenticationMechanism`, which manages OpenID Connect (OIDC) authorization code flow authentication, must build a correct redirect URL and set a state cookie.
```

Comment (rolfedh) [reply docs/src/main/asciidoc/security-proactive-authentication.adoc]: Updated. Thanks.

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
@rolfedh Is this one good to go or we are waiting for at least one more review from Michal or Sheila ?

**Review by sheilamjones — APPROVED** [review]
Hi @rolfedh,
Great job. I have no comments. All LGTM.

File: docs/src/main/asciidoc/security-proactive-authentication.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-proactive-authentication.adoc:21]:
```suggestion
You can change the default behavior if you only want to authenticate when the target page requires authentication.
```

File: docs/src/main/asciidoc/security-proactive-authentication.adoc
Comment (MichalMaler) [inline docs/src/main/asciidoc/security-proactive-authentication.adoc:33]:
```suggestion
If proactive authentication is used, accessing `SecurityIdentity` is a blocking operation.
```

**Review by MichalMaler — APPROVED** [review]

**Comment by sberyozkin** [comment]
@rolfedh Michal's suggestions look good to me, so please ask someone else from the team to merge it once they are applied

File: docs/src/main/asciidoc/security-proactive-authentication.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-proactive-authentication.adoc:21]: Updated. Thanks.

File: docs/src/main/asciidoc/security-proactive-authentication.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-proactive-authentication.adoc:33]: Updated. Thanks.

**Comment by rolfedh** [comment]
@geoand: For this team, what is the customary way to indicate that a PR is ready for merge? I want to make sure I do it correctly to avoid generating noise for the maintainers?

**Comment by geoand** [comment]
Usually when we know we are starting something that is work is progress, we make it a draft and then mark it as ready for review when we think we're dond

File: docs/src/main/asciidoc/security-proactive-authentication.adoc
Comment (rolfedh) [reply docs/src/main/asciidoc/security-proactive-authentication.adoc]: Updated. Thanks.

**Comment by rolfedh** [comment]
Thanks @geoand. I believe this document has been reviewed and is ready for merge. If I understand correctly, I'll wait for a maintainer to merge it. No action required on my part.

**Comment by geoand** [comment]
👌

**Pull request merged**
